pageserver/src/walingest.rs

```diff
@@ -373,6 +373,7 @@ impl WalIngest {
                 if info == pg_constants::XLOG_RUNNING_XACTS {
                     let xlrec = crate::walrecord::XlRunningXacts::decode(&mut buf);
                     self.checkpoint.oldestActiveXid = xlrec.oldest_running_xid;
+                    self.checkpoint_modified = true;
```

Comment:
Because otherwise we do not save checkpoint:
```
        // If checkpoint data was updated, store the new version in the repository
        if self.checkpoint_modified {
            let new_checkpoint_bytes = self.checkpoint.encode()?;

            modification.put_checkpoint(new_checkpoint_bytes)?;
            self.checkpoint_modified = false;
        }
```

Reply:
In principle, we can try to get oldestActiveXid not from checkpoint record in KV storage, but using its in-memory copy in walingest. But it seems to be non-trivial (most likely require moving it to timeline guarded by some lock).

Reply:
Makes sense, thanks